src/MPI/CMakeLists.txt

```diff
@@ -164,66 +161,63 @@ if (MPI_FOUND)
 
         set(lpf_cflags "${lpf_cflags} ${MPI_C_COMPILE_FLAGS} -I${mpi_include_flags} -fPIC" PARENT_SCOPE)
         set(lpf_lib_link_flags "${lpf_lib_link_flags} ${lib_lflags}" PARENT_SCOPE)
-        if (UNIX AND NOT APPLE)
-          set(lpf_exe_link_flags "${lpf_exe_link_flags} -rdynamic" PARENT_SCOPE)
-        endif()
     endforeach()
   
     include_directories(${MPI_C_INCLUDE_PATH})
     # add a test for dynamichook 
-    if (MPI_OPEN_PORT AND LPF_ENABLE_TESTS)
-        add_executable(dynamichook.t dynamichook.t.cpp dynamichook.cpp
-                $<TARGET_OBJECTS:${comlib}> )
-
-        target_include_directories(dynamichook.t PRIVATE ${GTEST_INCLUDE_PATH})
-        target_link_libraries(dynamichook.t ${MPI_C_LIBRARIES} gtest)
-        set_target_properties(dynamichook.t PROPERTIES
-                COMPILE_FLAGS "${MPI_C_COMPILE_FLAGS}"
-                LINK_FLAGS "${MPI_C_LINK_FLAGS}"
-        )
-        configure_file( dynamichook.t.sh.in dynamichook.t.sh @ONLY)
-        set( dynamic_hook_t_sh "${CMAKE_CURRENT_BINARY_DIR}/dynamichook.t.sh")
-        add_test(NAME dynamichook_1proc
-                 COMMAND bash ${dynamic_hook_t_sh} 1)
-        set_tests_properties( dynamichook_1proc PROPERTIES TIMEOUT 120 )
-        add_test(NAME dynamichook_2proc
-                 COMMAND bash ${dynamic_hook_t_sh} 2)
-        set_tests_properties( dynamichook_2proc PROPERTIES TIMEOUT 240 )
-        add_test(NAME dynamichook_3proc
-                 COMMAND bash ${dynamic_hook_t_sh} 3)
-        set_tests_properties( dynamichook_3proc PROPERTIES TIMEOUT 360 )
-        add_test(NAME dynamichook_10proc
-                 COMMAND bash ${dynamic_hook_t_sh} 10)
-        set_tests_properties( dynamichook_10proc PROPERTIES TIMEOUT 1200 )
+    if (NOT IS_OPENMPI AND LPF_ENABLE_TESTS)
+        add_gtest(dynamichook.t "mpimsg" ON
+            ${CMAKE_CURRENT_SOURCE_DIR}/dynamichook.t.cpp 
+            ${CMAKE_CURRENT_SOURCE_DIR}/dynamichook.cpp 
+            ${CMAKE_CURRENT_SOURCE_DIR}/mpilib.cpp)
+        
+            configure_file( dynamichook.t.sh.in dynamichook.t.sh @ONLY)
+            set( dynamic_hook_t_sh "${CMAKE_CURRENT_BINARY_DIR}/dynamichook.t.sh")
+            add_test(NAME dynamichook_1proc
+                COMMAND bash ${dynamic_hook_t_sh} 1)
+            set_tests_properties( dynamichook_1proc PROPERTIES TIMEOUT 30 )
+            add_test(NAME dynamichook_2proc
+                COMMAND bash ${dynamic_hook_t_sh} 2)
+            set_tests_properties( dynamichook_2proc PROPERTIES TIMEOUT 30 )
+            add_test(NAME dynamichook_3proc
+                COMMAND bash ${dynamic_hook_t_sh} 3)
+            set_tests_properties( dynamichook_3proc PROPERTIES TIMEOUT 30 )
+            add_test(NAME dynamichook_10proc
```

Comment:
add property `RUN_SERIAL TRUE`  to each dynamichook test property to force serial execution when Ctest is run with multiple parallel jobs. This way, these tests won't deadlock when run in parallel (that's because they all try to connect throught the same TCP port)

Reply:
I have done it, but not tested it.